Chore/clean up housing cards

BJHunnicutt opened the pull request:
More updates to Housing 2019 Cards. Still a 🚧work in progress 🚧, but there are a lot of good intermediate changes I wanted to get feedback on before I continue.

BJHunnicutt committed:
Update heat map, try adding ADU points, add alternate map option

BJHunnicutt committed:
Clean up home appreciation map

BJHunnicutt committed:
Revert bad yarn.lock changes

BJHunnicutt committed:
Use color constants for housing card map

BJHunnicutt committed:
Start pulling housing map data into Card 1

BJHunnicutt committed:
Clean up Housing Card 4 with new Compare map & common color scale

BJHunnicutt committed:
Update housing card 1 - Black Population Displacement - to include a map & more thresholds

BJHunnicutt committed:
Merge branch 'master' into chore/clean-up-housing-cards

BJHunnicutt committed:
Add protect prop to charts to avoid color issues

BJHunnicutt committed:
Merge branch 'chore/clean-up-housing-cards' of https://github.com/hackoregon/civic into chore/clean-up-housing-cards

BJHunnicutt committed:
Add a temporary full card loading indicator (temp b/c I assume it will move to a central location

BJHunnicutt committed:
Add a full-card skeleton loading indicator to all Housing cards

BJHunnicutt committed:
Remove helper text default placeholder on radio component

BJHunnicutt committed:
Remove unused prop on RadioGroup that is causing a console error

BJHunnicutt committed:
Clean up Housing Card 2

BJHunnicutt committed:
Update Housing card 4 todos

BJHunnicutt committed:
Remove unused ADU maps/plots

BJHunnicutt committed:
Update formatting of MapTooltip to allow for more information

BJHunnicutt committed:
Add more data to Housing Card 2 map tooltips

Dianna commented:
Card #2: Racial Demographics of Home Loan Approvals
- Home loan approval rates needs a legend (what is green vs brown?)
- Maybe set a max zoom in / out?

Card #4: Black Population Share Change Over Time
- Slider button could be more obvious
- I think a 1 line explanation would be good (i.e., Slide to compare 1990 to 2017)
- Maybe set a max zoom in / out?

Card #6:
- Maybe set a max zoom in / out?
- Top 2 maps are the same data right, just different implementations? I like the top one, with tooltips
- Bottom visualization is scatterplot without enough data or line graph without a line?

Card #8:
- Make sure to add legend when this map is available

Card #10: Households of Color Earn Substantially Less than White Households
- Dot colors match lines now. Was a chart implementation issue

Dianna commented on packages/2019-housing/src/components/BlackPopulationChange/BlackPopulationChangeVisualization.js:
```diff
@@ -5,13 +5,18 @@ import { scaleQuantize, extent } from "d3";
 
 import {
```
In `packages/2019-housing/src/components/AduDistributions/index.js`, the eslint-disable is not working for init, needs to be repositioned

Dianna commented:
In `AduDistributions/index.js`, `HolcRedlining/index.js`, `HomeAppreciation/index.js`, `HomeLoanApprovals/index.js`, `HomeOwnershipRates/index.js`, `HouseholdIncomeByRace/index.js`, and `HousingDisplacement/index.js` the eslint-disable is not working for init, needs to be repositioned.

Or this works:

```
useEffect(() => {
    init();
  }, [
    /*
    https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

    Add second argument to prevent useEffect running init() again
    */
  ]);
```

BJHunnicutt committed:
Wire up API to pull in real redlining data

BJHunnicutt committed:
Make Housing redlining map card

BJHunnicutt committed:
Fix init() issue in housing card index.js files

BJHunnicutt committed:
Remove todos from Housing Card 10

BJHunnicutt committed:
Set Min/Max zoom on Housing card maps

BJHunnicutt commented:
@Dianna, I think I addressed most of your comments, I updated the init() thing everywhere and added min/max zoom to all the maps.

Card 2: My next task is to build a MapLegend (which will affect most of the cards), but I want to get these changes in before I do that. 

Card 4: I am planning to work a bit more on the compare map to make the slider more obvious, but in the meantime I added what you recommended to the subtitle of the map.

Card 6: Housing had asked me to make both maps & they also picked the top 1. The scatter plot was just a stand-in/fake data because the API wasn't ready yet, but I removed it for now.

(**Side note**: If you test this again, Card 2 will infinitely load because they are changing the API)

BJHunnicutt committed:
Fix init() issue in housing card index.js files

BJHunnicutt committed:
Merge branch 'master' into chore/clean-up-housing-cards

BJHunnicutt committed:
Get those darn inits right

BJHunnicutt committed:
Update NcdbYearlyData API calls to use new API changes

BJHunnicutt committed:
Add slider map instruction

BJHunnicutt committed:
Fix bar chart in housing SC9

BJHunnicutt committed:
Make map in housing SC9 have an interactive threshold controlled by a slider

BJHunnicutt committed:
Add area plot behind SC9 line chart to show confidence intervals

BJHunnicutt committed:
Fix console errors caused by tooltips

BJHunnicutt committed:
Use a more civic-esque green for diverging color scale on card 2

jaronheard reviewed (approved):
If you want to merge this so that others can see to review, go ahead when you please :)
